hack/lib/util.sh

```diff
@@ -139,6 +135,23 @@ kube::util::gen-doc() {
   ${cmd} "${tmpdir}"
   # create the list of generated files
   ls "${tmpdir}" | LC_ALL=C sort > "${tmpdir}/.files_generated"
+
+  # remove all old generated file from the destination
+  for file in $(cat "${dest}/.files_generated" 2>/dev/null); do
+    if [[ -e "${tmpdir}/${file}" && ! -z "${skipline}" ]]; then
+      # If RCS format diff has 3 lines, only 1 line changed.
+      local fdiff=$(diff -n "${dest}/${file}" "${tmpdir}/${file}")
+      if [[ $(echo "${fdiff}" | wc -l) == 3 && $(echo "${fdiff}" | grep -e "${skipline}") ]]; then
```

Comment:
`${skipline}` is just a general grep regexp. The purpose of this is to detect files where the generated output does no meaningful change, it just updates the timestamp
```
-###### Auto generated by spf13/cobra at 2015-04-17 18:59:11.486604006 +0000 UTC
+###### Auto generated by spf13/cobra at 2015-04-23 20:38:18.472875747 +0000 UTC
```